src/LibraryManager/Configuration/Settings.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Web.LibraryManager.Contracts.Configuration;
+using Microsoft.Web.LibraryManager.Utilities;
+
+namespace Microsoft.Web.LibraryManager.Configuration
+{
+    internal class Settings : ISettings
+    {
+        /// <summary>
+        /// File name where user settings (not project settings) are stored
+        /// </summary>
+        private const string ConfigFileName = "libman.config";
+
+        private XDocument _configFile;
+
+        // protected for tests to override
+        protected virtual string ConfigFilePath => Path.Combine(UserDataRoot, ConfigFileName);
+
+        public static Settings DefaultSettings { get; } = new Settings();
+
+        public Settings()
+        {
+
+            if (File.Exists(ConfigFilePath))
+            {
+                try
+                {
+                    _configFile = XmlUtility.LoadXmlDocument(ConfigFilePath);
+                }
+                catch (XmlException)
```

Comment:
Given that the file exists, if there were a different type of file exception (file lock maybe?) I don't think they should be suppressed here since there isn't a good recovery strategy.  I.e. if I can't read from it, I probably can't use `InitSettingsFile()` to write to it; and then the Settings instance being constructed is not in a valid state.  Seems better not to catch that case here.

Reply:
I guess the question is how it propagates to the user and what the end result is. The CLI will just exist with an exception, which might not be horrible. It would be worse if we crashed VS or GUI, but we don't support it there yet, so I'll this up to you.